src/main/java/uk/gov/companieshouse/insolvency/data/service/InsolvencyService.java

```diff
@@ -1,27 +1,27 @@
 package uk.gov.companieshouse.insolvency.data.service;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.ObjectWriter;
 import org.springframework.stereotype.Service;
-import uk.gov.companieshouse.api.delta.Insolvency;
-import uk.gov.companieshouse.insolvency.data.requests.InsolvencyRequest;
+import uk.gov.companieshouse.api.insolvency.InternalCompanyInsolvency;
+import uk.gov.companieshouse.insolvency.data.InsolvencyRepository;
+import uk.gov.companieshouse.logging.Logger;
 
 @Service
 public class InsolvencyService {
 
-    public InsolvencyService() {
+    private final Logger logger;
+    private final InsolvencyRepository insolvencyRepository;
+
+    public InsolvencyService(Logger logger, InsolvencyRepository insolvencyRepository) {
+        this.logger = logger;
+        this.insolvencyRepository = insolvencyRepository;
     }
 
     /**
-     * Save insolvency service layer method.
-     *
-     * @param  companyNumber  companyNumber
-     * @param  insolvency  the insolvency request data
+     * Persist company insolvency information to mongodb collection.
+     * @param insolvencyApi company insolvency information {@link InternalCompanyInsolvency}
      */
-    public void saveInsolvency(
-            String companyNumber,
-            InsolvencyRequest insolvency) throws JsonProcessingException {
-        // TODO Save to database
+    public void saveInsolvency(String companyNumber, InternalCompanyInsolvency insolvencyApi) {
+        insolvencyRepository.save(insolvencyApi);
```

Comment:
I am not sure we can save internal insolvency directly, we need data object matching MongoDB structure if my understanding is correct. Lets connect offline